app/controllers/requests_controller.rb

```diff
@@ -0,0 +1,21 @@
+class RequestsController < ApplicationController
+
+  def index
+    @requests = current_user.requests
+  end
+
+  def Show
+
+  end
+
+  def new
+    @request = Request.create(request_params)
+  end 
+
+  def create
+    book = Book.find(params[:book_id])
+    @request = book.requests.create(user_id: current_user.id)
+    @book_id = params[:book_id]
```

Comment:
Suggestion: make `book` an instance variable (`@book`) and delete `@book_id`

Remember to use `@book.id` in `create.js.erb`